pydantic/env_settings.py

```diff
@@ -232,7 +235,16 @@ def explode_env_vars(self, field: ModelField, env_vars: Mapping[str, Optional[st
             env_var = result
             for key in keys:
                 env_var = env_var.setdefault(key, {})
-            env_var[last_key] = env_val
+
+            if env_val:
+                try:
+                    env_val = settings.__config__.json_loads(env_val)
+                except JSONDecodeError:
```

Comment:
```suggestion
                except ValueError:
```
More consistent to just use `ValueError`